### PR TITLE
feat(config): add Minoston MP31ZP

### PR DESCRIPTION
**Pull request by boomam**
Used MP21ZP as template json.  
&nbsp;  
Attempted to copy/note details from both manufacturer manual:  
https://products.z-wavealliance.org/ProductManual/File?folder=&filename=product_documents/4626/MP31ZPmanual-20220428.pdf  
&nbsp;  
and Z-Wave database:  
https://products.z-wavealliance.org/products/4626?selectedFrequencyId=-1).  
&nbsp;  
Configuration capabilties not attempted, but appears to work fine with the MP21 config on current installation.

PR description here

### CODE REVIEW

**Review by AlCalzone — CHANGES_REQUESTED** [review]
A short look into the manual shows that the config parameters are completely different:
grafik
(and there are 17 total, not only 8)

**Comment by boomam** [comment]
> A short look into the manual shows that the config parameters are completely different: grafik (and there are 17 total, not only 8)

Fair.  
I have no idea how to reference or add these to the json, but if we want that correcting, someone else will need to add it.

Happy to help document the process and add to the docs to help others add though if we can see some examples of what/how.

**Comment by AlCalzone** [comment]
> I have no idea how to reference or add these to the json

Does the documentation [here](https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=paraminformation) and [here](https://zwave-js.github.io/node-zwave-js/#/config-files/contributing-files) help?
If so, make sure to follow the [style guide](https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide).

Theres also a section on [importing templates](https://zwave-js.github.io/node-zwave-js/#/config-files/using-templates), but I can take care of it if this is too much.

**Comment by boomam** [comment]
> > I have no idea how to reference or add these to the json
> 
> Does the documentation [here](https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=paraminformation) and [here](https://zwave-js.github.io/node-zwave-js/#/config-files/contributing-files) help? If so, make sure to follow the [style guide](https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide).
> 
> Theres also a section on [importing templates](https://zwave-js.github.io/node-zwave-js/#/config-files/using-templates), but I can take care of it if this is too much.

I'll take another look at lunch, but I am not hopeful, as I looked the other day and it wasn't clear.
I'll report back later today.

**Comment by boomam** [comment]
ok, I've taken a quick look at this and added as best I can.  
I am unsure on the multi-channel settings for the associations groups, but otherwise the whole block of json passes checks/linting on VSCode/a few others, so assuming the content is fine, the only issue would be any linting specific to this platform

**Comment by boomam** [comment]
@zwave-js-bot fix lint

**Comment by boomam** [comment]
I've played around with correcting some of the errors listed, but as with GitHub actions, its sometimes like shooting in the dark trying to diagnose.  
I'll attempt to see if the bot can fix whats wrong, but I can't dedicate any more time to this today unfortunatelly.

**Comment by boomam** [comment]
@zwave-js-bot fix lint

**Comment by AlCalzone** [comment]
You have syntax errors, which the bot cannot fix. Missing commas, additional brackets.

I suggest using vscode or our hosted gitpod instance which both highlight these errors.

**Comment by boomam** [comment]
> You have syntax errors, which the bot cannot fix. Missing commas, additional brackets.
> 
> I suggest using vscode or our hosted gitpod instance which both highlight these errors.

I did, vscode didn't flag anything up for me, neither did an online checker I often use.
I won't be looking at this for at least a week or so most likely, so if the syntax needs fixing before that, someone else will need to correct.

I'll try and find 30mins when I get home later, and test with a different set of linters to be triple sure, but otherwise it'd have to wait.

**Comment by boomam** [comment]
@AlCalzone I can see from the latest log that its again complaining about something to do with the device config being malformed.  
&nbsp;  
I have ran this through a few more linters and only found 2x missing commas prior to this - the JSON is 100% sound.  
&nbsp;  
Whatever checks are being ran, I am going to need **you** to diagnose to get this completed & merged.
Additionally, I'd suggest checking the manual for the device (linked above) as I'm not sure what's been entered will work or not regardless of JSON formatting - for some who's never touched this before, I dont think its a great idea to have it merged for usage without sense checking.   
&nbsp;  
Based on it now linting across 4x platforms 100% fine, I will no longer be touching this PR.

**Comment by AlCalzone** [comment]
Alright, I'll check it out

**Comment by boomam** [comment]
> Alright, I'll check it out

Thanks!  :-)

File: packages/config/config/devices/0x0312/mp31zp.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
+{
+	"manufacturer": "Minoston",
+	"manufacturerId": "0x0312",
+	"label": "MP31ZP",
+	"description": "Z-Wave Plug with Energy Monitor",
+	"devices": [
+		{
+			"productType": "0xff00",
+			"productId": "0xff0e",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0312/mp31zp.json:9]: @boomam This information is entered in the Z-Wave Alliance DB, but it seems to duplicate the MP21ZP IDs. I'm guessing this is an error in the DB entry.

Can you post the device ID that is shown for your device? In zwavejs2mqtt, you'll find it when you click on the device in the device list. In HA I'm not sure, but the debug information for the node should probably contain it.

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/config/config/devices/0x0312/mp31zp.json (hunk 1 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0312/mp31zp.json:9]: The rest is cleaned up, so I can merge this when the correct IDs are known.

**Comment by AlCalzone** [comment]
Somehow my comment disappeared... For future reference: Your mistake was that you entered strings (values in quotes) for several values that expected a plain number or `true` / `false`. VSCode does highlight this for me, so I'm not sure if there is something missing in the default configuration.

File: packages/config/config/devices/0x0312/mp31zp.json (hunk 1 above)
Comment (boomam) [reply packages/config/config/devices/0x0312/mp31zp.json:9]: 786-65294-65280 (0x0312-0xff00-0xff0e)

**Comment by boomam** [comment]
> Somehow my comment disappeared... For future reference: Your mistake was that you entered strings (values in quotes) for several values that expected a plain number or `true` / `false`. VSCode does highlight this for me, so I'm not sure if there is something missing in the default configuration.

Device ID added (replied above) -  

RE: strings - not sure, a few linters confirmed the code was fine.  
Perhaps an extension you are using?

**Comment by AlCalzone** [comment]
> not sure, a few linters confirmed the code was fine.

It **was** valid JSON, which is what most linters check for, it's just not a valid device configuration file. That's why we have an extra JSON Schema and additional checks for validation. And that's why we recommend VSCode or the hosted Gitpod instance for working on config files.

> 786-65294-65280 (0x0312-0xff00-0xff0e)

Hmm that IS identical with the MP21ZP. Any chance you have a 21 too, so we can compare firmware versions? I sure hope there is any way at all to distinguish them.

**Comment by boomam** [comment]
> > not sure, a few linters confirmed the code was fine.
> 
> It **was** valid JSON, which is what most linters check for, it's just not a valid device configuration file. That's why we have an extra JSON Schema and additional checks for validation. And that's why we recommend VSCode or the hosted Gitpod instance for working on config files.
> 
I used VSCode ;-)  

> > 786-65294-65280 (0x0312-0xff00-0xff0e)
> 
> Hmm that IS identical with the MP21ZP. Any chance you have a 21 too, so we can compare firmware versions? I sure hope there is any way at all to distinguish them.

I unfortunatelly do not, no.  
Anything else we could test/differentiate with with the 31ZP to help?   
Or, is there a way to add the 31ZP to the database, but have it be able to be manually selected in the GUI? I can't imagine this is the first manufacturer to do something odd?

**Comment by AlCalzone** [comment]
> I used VSCode ;-)

Ok then I don't understand whats wrong. Even the Github-embedded VSCode (press <kbd>.</kbd>) complains:
grafik

> Anything else we could test/differentiate with with the 31ZP to help?

You could try contacting the manufacturer and ask them how they can be distinguished. Sometimes its just the firmware version range for example.

> Is there a way to add the 31ZP to the database, but have it be able to be manually selected in the GUI

No.

**Comment by boomam** [comment]
> You could try contacting the manufacturer and ask them how they can be distinguished. Sometimes its just the firmware version range for example.
> 
Not sure i'll have time to spend time going through bureaucratic phone IVRs or emailing support teams who have no clue as to the question...  
Any other options?

**Comment by AlCalzone** [comment]
> Any other options?

No clue. At this point I'm afraid only the manufacturer or someone with both devices will be able to tell us.

**Comment by boomam** [comment]
> > Any other options?
> 
> No clue. At this point I'm afraid only the manufacturer or someone with both devices will be able to tell us.

Then lets do this, I'll post a question on reddit/the HA forum, see if anyone has it, and if no one replies by the end of the month, I'll just buy one (as I need a second outlet anyway).  
I will post whatever links for reference here, once I do.

Specifically, you need what info off another owner of a MP21ZP? 
I want to make it as easy as possible for them to grab what's needed.

**Comment by AlCalzone** [comment]
> you need what info off another owner of a MP21ZP?

Firmware version, as shown in zwavejs2mqtt in the **FW** column. HomeAssistant probably has this info somewhere too.

**Comment by boomam** [comment]
> > you need what info off another owner of a MP21ZP?
> 
> Firmware version, as shown in zwavejs2mqtt in the **FW** column. HomeAssistant probably has this info somewhere too.

ok.
Posted - 
https://community.home-assistant.io/t/minoston-mp21zp-firmware-information/448258
https://www.reddit.com/r/homeassistant/comments/wj8tk5/minoston_mp21zp_firmware_information/?

I'll keep an eye on the threads and report back any findings on the 21.

For the 31, anything i can provide you to help diagnose that aspect, or do you have everything needed until you get firmware info for the 21?

**Comment by AlCalzone** [comment]
You can show me your firmware version

**Comment by boomam** [comment]
> You can show me your firmware version

List on the summary as v1.0
Expanding 'version' under the node, shows - 

Z-Wave protocol version: 7.13
Z-Wave chip firmware versions: "1.0"
Z-Wave chip hardware version: 1
SDK version: 7.13.10
Z-Wave application framework API version: 10.13.10
Z-Wave application framework API build number: 423
Serial API version: unused
Serial API build number: 0
Z-Wave protocol version: 7.13.10
Z-Wave protocol build number: 423
Application version: 1.0.1
Application build number: 43707

**Comment by boomam** [comment]
We have a reply on the HA forum: 
https://community.home-assistant.io/t/minoston-mp21zp-firmware-information/448258
Formatted below, highlighting differences.

Z-Wave protocol version: 7.13
Z-Wave chip firmware versions: "1.0"
Z-Wave chip hardware version: 1
**SDK version: 7.13.9**
**Z-Wave application framework API version: 10.13.9**
**Z-Wave application framework API build number: 405**
Serial API version: unused
Serial API build number: 0
**Z-Wave protocol version: 7.13.9**
**Z-Wave protocol build number: 405**
Application version: 1.0.1
Application build number: 43707

**Comment by AlCalzone** [comment]
Hmm that's bad. The differences are only because they used a different version of the Z-Wave SDK to build their firmware and we're not equipped to use that info to distinguish devices. 

Grasping at straws here: Any chance the config parameters in the manual are wrong and they just reused the 21 firmware?

**Comment by boomam** [comment]
> Grasping at straws here: Any chance the config parameters in the manual are wrong and they just reused the 21 firmware?

Absolutely no idea on that one.
Would seem weird that the z-wave database spec shows extra variables, and the firmware shows the same on the device though.

**Comment by AlCalzone** [comment]
Can you check if the config parameters 9 - 17 actually exist? If not, can you check if the parameters 1-8 behave as the ones from MP21ZP: https://devices.zwave-js.io/?jumpTo=0x0312:0xff00:0xff0e:0.0

You'll likely need to use the custom config parameter function in zwavejs2mqtt to test this:
grafik
Take values and value size from the 31 manual you linked. For the 21 params you can just use the ones available in the UI.

Setting a value (that is out of range for the 21) and then reading it should yield the same value you set. If not, there might be discrepancy.

**Comment by boomam** [comment]
> Can you check if the config parameters 9 - 17 actually exist? If not, can you check if the parameters 1-8 behave as the ones from MP21ZP: https://devices.zwave-js.io/?jumpTo=0x0312:0xff00:0xff0e:0.0
> 
> You'll likely need to use the custom config parameter function in zwavejs2mqtt to test this: grafik Take values and value size from the 31 manual you linked. For the 21 params you can just use the ones available in the UI.
> 
> Setting a value (that is out of range for the 21) and then reading it should yield the same value you set. If not, there might be discrepancy.

I'm testing this on the 31 or on the 21 from the poster in the HA forum?  
What are we looking for, to test out of range? An error of some sort?

**Comment by AlCalzone** [comment]
Test on the 31 please. I'm looking for timeouts on reading params 9-17 or params 1-8 that behave like the 21 variant instead what the 31 manual says. If thats the case, maybe the 21 and 31 are the same and they messed up in the manual. If not and they exist and behave like they should according to the 31 manual, then I'm afraid there's nothing we can do here for now until they fix the device identification.

**Comment by boomam** [comment]
ok, but what am i looking for _specifically_?
Should an error display if i send a parameter in the 9-17 range, and the firmware is actually firmware for the 21?
Or will nothing happen at all? In which case, how do we know if its worked?
I plan to test #17, as right now the switch is powering something i need on until later today, and #17 seems low impact.

**Comment by boomam** [comment]
I sent an '03', with parameter 17, to turn off the LED on the device.  
Doesnt appear to have done anything on the device itself.  
The log shows it did successfully send, but it sent a '3' and not an '03' like the manual specifies - 
```
2022-08-08 10:43:11.213 INFO ZWAVE: Calling api sendCommand with args: [
{ nodeId: 2, commandClass: 112 },
'set',
[ 17, 3, 1, [length]: 3 ],
[length]: 3
]
```

**Comment by AlCalzone** [comment]
Now perform a GET on param 17, and while you do so capture a [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug`.

**Comment by boomam** [comment]
Pretty much the same in the log, no additional data - 
```
2022-08-08 10:46:41.168 INFO ZWAVE: Calling api sendCommand with args: [
{ nodeId: 2, commandClass: 112 },
'get',
[ 17, [length]: 1 ],
[length]: 3
]
```

**Comment by AlCalzone** [comment]
Wrong log ;)

**Comment by boomam** [comment]
So the logs dont show the logs? Interesting... ;-)

What am i looking for specifically in the log export?
The only thing i see is near identical to the other log - 
```
2022-08-08 10:46:41.168 DEBUG SOCKET: Event ZWAVE_API emitted to cIYuFNpfBHAFVH6OAAAL
2022-08-08 10:46:41.169 INFO ZWAVE: Calling api sendCommand with args: [
  { nodeId: 2, commandClass: 112 },
  'get',
  [ 17, [length]: 1 ],
  [length]: 3
]
2022-08-08 10:46:41.204 DEBUG MQTT: Publishing to zwave/Server_Room/Network_Rack/112/0/1: { time: 1659970001203, value: 0 } with options { qos: 1, retain: true }
2022-08-08 10:46:41.204 INFO ZWAVE: Node 2: value updated: 112-0-1 0 => 0
```

**Comment by AlCalzone** [comment]
Please read the link I posted again. You're looking at the entirely wrong log (zwavejs2mqtt/application logs, not the driver logs)

**Comment by boomam** [comment]
We really need to get these logs labeled...

Anyway, the other log doesn't show the same time values (as in, looks to be a different time zone, with a drift too) as the main log I downloaded, so its difficult to line up time wise. 
Looking through, I'm not seeing anything where a command has been sent to the device.
To check, is there a specific wording pattern im looking for?

**Comment by AlCalzone** [comment]
Its in UTC, not local time. Probably best if you have the log open in one window, then perform a SET and GET in another and post the log window contents.

**Comment by boomam** [comment]
> Its in UTC, not local time.  

I'm aware, but there's still drift compared to the other log. ;-)  
&nbsp;  

> Probably best if you have the log open in one window, then perform a SET and GET in another and post the log window contents.  

Tried that, the number of energy reporting logs means its still a manual process looking through thousands of lines.
Is there a specific word pattern i should be looking for?

**Comment by AlCalzone** [comment]
> the number of energy reporting logs means its still a manual process looking through thousands of lines.

Uhh, how many reports per second are you getting? This doesn't seem [healthy](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=optimizing-the-reporting-configuration), especially if you deleted the logfile beforehand.

> Is there a specific word pattern i should be looking for?

Try `ConfigurationCC`

**Comment by boomam** [comment]
> Uhh, how many reports per second are you getting? This doesn't seem [healthy](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=optimizing-the-reporting-configuration), especially if you deleted the logfile beforehand.  

Seemingly 1 per second.

 
> Try `ConfigurationCC`

```
2022-08-08T14:46:41.173Z DRIVER » [Node 002] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      3
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 19
                                    └─[ConfigurationCCGet]
                                        parameter #: 17
2022-08-08T14:46:41.180Z SERIAL « [ACK]                                                                   (0x06)
2022-08-08T14:46:41.180Z SERIAL « 0x010401a90152                                                       (6 bytes)
2022-08-08T14:46:41.181Z SERIAL » [ACK]                                                                   (0x06)
2022-08-08T14:46:41.181Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2022-08-08T14:46:41.194Z SERIAL « 0x011d00a90300000100c87f7f7f7f00000300000000030100007f7f7f7f7fff    (31 bytes)
2022-08-08T14:46:41.194Z SERIAL » [ACK]                                                                   (0x06)
```

**Comment by AlCalzone** [comment]
> Seemingly 1 per second.

Bit much if you ask me.

The log isn't complete, is it? I'm missing the callback (definitely) and the device response (or not)

**Comment by boomam** [comment]
> Bit much if you ask me.

I agree, but according to the configuration, the interval is set to 5 minutes.

> The log isn't complete, is it? I'm missing the callback (definitely) and the device response (or not)

The next part of the log after that is just an energy report.

**Comment by AlCalzone** [comment]
> I agree, but according to the configuration, the interval is set to 5 minutes.

Defaults for timed reports are rarely good.

> The next part of the log after that is just an energy report.

Just give me everything starting from the outgoing command you quoted above.

**Comment by boomam** [comment]
> Defaults for timed reports are rarely good.

I thought the default, 5mins to be more than good enough. It just doesnt seem to be taking that default...
Tried manually set it with a custom config to 30 seconds, still reports every second or less.

> Just give me everything starting from the outgoing command you quoted above.  

Can you be more specific please?  
I have given you the lines where the command starts and ends already?

**Comment by AlCalzone** [comment]
> I have given you the lines where the command starts and ends already?

You didn't. Just start the log, send both commands, wait a few seconds after each, then give me the entire log you have at this point.

**Comment by boomam** [comment]
> > I have given you the lines where the command starts and ends already?
> 
> You didn't. Just start the log, send both commands, wait a few seconds after each, then give me the entire log you have at this point.

I can assure you I did.  
I'm just going to upload the whole log. Give me a few mins.

**Comment by boomam** [comment]
https://xxx.xxx.com

Let me know when you have it and i'll nuke the share/log off my NC.

**Comment by AlCalzone** [comment]
That is asking me for a username and password

**Comment by boomam** [comment]
Really, one sec, let me adjust the API passthrough.

**Comment by AlCalzone** [comment]
Something is loading now, but I can't press the login button

**Comment by AlCalzone** [comment]
```
2022-08-08T14:46:41.173Z DRIVER » [Node 002] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      3
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 19
                                    └─[ConfigurationCCGet]
                                        parameter #: 17
...
2022-08-08T14:46:41.205Z DRIVER « [Node 002] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -56 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 93
                                    └─[ConfigurationCCReport]
                                        parameter #: 1
                                        value size:  1
                                        value:       0
```

This is exactly what I was looking for. You request param 17 and the device says "no such thing, here's my first one, number 1".

So the questions are now:
- Do any of params 9-16 exist?
- Do params 1-8 behave like on the MP21 or as described in the MP31 manual? To test you can look at parameters that have a larger range on the MP31, set values that are out of range for the MP21, read them back and see if the same value you set gets reported. Also the device behavior might change differently than described.

**Comment by boomam** [comment]
I'll have to test other parameters later today, as its currently plugged into my router.  
Right now, if we don't think that other testing will yield a successful config, I'll help complete the testing above, but I'm just gonna send the thing back to Amazon after and get a different model.

**Comment by AlCalzone** [comment]
My hope right now is that we identify the firmware and its config parameters to be identical to the MP21, so we don't need the extra file.

**Comment by boomam** [comment]
ok.  
If i can get the electricity wattage reporting to not be every second or so (no idea why its doing it), then later today i'll run through ALL the parameters, dump the log, and send over.

**Comment by AlCalzone** [comment]
Honestly I don't mind the extra reports

**Comment by boomam** [comment]
# MP31ZP Parameter Testing
> Log File sent to @AlCalzone via Discord.

## Useful URLs
[Manufacturer Provided Product Manual](https://products.z-wavealliance.org/ProductManual/File?folder=&filename=product_documents/4626/MP31ZPmanual-20220428.pdf)
[Z-Wave Alliance Product Page](https://products.z-wavealliance.org/products/4626?selectedFrequencyId=-1)

## Notes
* Log Level = Debug
* Log wiped at beginning of testing
* GET commands ran after each 'SET'
* 5 seconds delay between 'SET' and 'GET'.
* 30 seconds after last command, logging returned to normal/off & log exported.

## Parameter Testing
| Parameter | Value Sent | Expected Outcome | Outcome (IRL / in GUI) |
| ----- | ----- | ----- | ----- |
| 1 - Restores state after power failure | 02 | Power Off | IRL - Power Remains on after taking off outlet and putting back |
| 2 - Power Wattage Report Value Threshold(Wattage) | 20 | 20w | No indicators seen in either GUI or IRL |
| 3 - Power Wattage Report Frequency (Timer) | 45 | 45 seconds | GUI - Shows Interval as 45  |
| 4 - Energy (kWh) Report Frequency (Timer) | 45 | 45 seconds | No indicators seen in either GUI or IRL  |
| 5 - Voltage (V) Report Frequency (Timer) | 45 | 45 seconds | No indicators seen in either GUI or IRL |
| 6 - Electricity (A) Report Frequency (Timer) | 45 | 45 seconds | No indicators seen in either GUI or IRL  |
| 7 - Overload Protection (Current) | 12 | 12A |  No indicators seen in either GUI or IRL  |
| 8 - Endpoint1 Enable Auto Turn-OFF Timer | 1 | Auto Turn Off =on | No indicators seen in either GUI or IRL  |
| 9 - Endpoint1 Count Down Configuration(Turn off the output by time) | 30 | 30 minutes | No indicators seen in either GUI or IRL  |
| 10 - Endpoint1 Enable Auto Turn-ON Timer | 1 | Auth Turn Off =on | No indicators seen in either GUI or IRL  |
| 11 - Endpoint1 Count Down Configuration(Turn on the output by time) | 30 | 30 minutes | No indicators seen in either GUI or IRL  |
| 12 - Endpoint2 Enable Auto Turn-OFF Timer | 1 | Auth Turn Off =on | No indicators seen in either GUI or IRL  |
| 13 - Endpoint2 Count Down Configuration(Turn off the output by time) | 30 | 30 minutes | No indicators seen in either GUI or IRL  |
| 14 - Endpoint2 Enable Auto Turn-ON Timer | 1 | Auth Turn Off =on | No indicators seen in either GUI or IRL  |
| 15 - Endpoint2 Count Down Configuration(Turn on the output by time) | 30 | 30 minutes | No indicators seen in either GUI or IRL  |
| 16 - Manual Control | 0 | Manual Control Disabled | No indicators seen in either GUI or IRL  |
| 17 - Status LED Configuration | 03 | LED is always Off | IRL - LED Turned Off |

**Comment by boomam** [comment]
Unrelated, but setting "[2-112-0-6] Power Reporting Interval" to something else, then back to '1', seems to have slowed down the energy reporting across all 4x types to once per minute according to the log.
Strange!

**Comment by AlCalzone** [comment]
Ok, so neither config parameter starting with 9 seems to exist. 4 didn't accept the value 45 and reverted to 2. 5 didn't either and reverted to 5. 7 didn't accept the 12 and reverted to 1.

This looks suspiciously like the parameters and ranges for the MP21ZP:
- param 4, default 2 = turn on after power loss
- param 5 = power reporting threshold, max is 5
- param 7 = current reporting threshold, max is 10

**Comment by AlCalzone** [comment]
Closing in favor of #4894

**Pull request closed without merging**